src/mgos_pppos.c

```diff
@@ -941,13 +964,13 @@ bool mgos_pppos_disconnect(int if_instance) {
       pd->pppcb = NULL;
       pppapi_close(pppcb, 0 /* no_carrier */);
     }
-    pd->cmd_error_state = PPPOS_IDLE;
-    pd->cmd_success_state = PPPOS_IDLE;
-    // If we are not running a user command, go to IDLE immediately,
+    pd->cmd_error_state = PPPOS_CLOSING;
+    pd->cmd_success_state = PPPOS_CLOSING;
+    // If we are not running a user command, go to CLOSING immediately,
     // otherwise finish the command sequence.
     if (pd->state == PPPOS_INIT || pd->state == PPPOS_START_PPP ||
```

Comment:
actually, this needs an update.please add a switch statement with cases for each existing state:

PPPOS_IDLE -> no change
  PPPOS_INIT -> IDLE
  PPPOS_START_SEQ -> IDLE
  PPPOS_RESET -> IDLE
  PPPOS_RESET_HOLD -> IDLE
  PPPOS_RESET_WAIT -> IDLE
  PPPOS_BEGIN_WAIT -> IDLE
  PPPOS_BEGIN -> IDLE
  PPPOS_SETUP -> IDLE
  PPPOS_CMD -> IDLE/CLOSING
  PPPOS_CMD_RESP -> IDLE/CLOSING
  PPPOS_START_PPP -> IDLE/CLOSING
  PPPOS_RUN -> IDLE/CLOSING
  PPPOS_CLOSING - no change

IDLE/CLOSING: go to CLOSING if pppapi_close was invoked (and we expect a callback), otherwise IDLE.

Reply:
This is for `pd->cmd_error_state` and `pd->cmd_success_state`? Just to make sure